src/packages/storage.ts

```diff
@@ -5,21 +5,12 @@ const log = getLogger(__filename);
 // Storage service abstraction
 /* istanbul ignore next */
 export abstract class StorageService {
+    public isSync: boolean = true;
     public abstract setItem = (key: string, value: string): void => undefined;
     public abstract getItem = (key: string): string | null => null;
+    public abstract async getItemAsync(key: string): Promise<string | null>;
```

Comment:
can modify StorageService to always return Promise and assume to be async. Will reduce the requirement of isSync property and StorageHelper class. You can evaluate.